### PR TITLE
Day picker accessibility

### LINKED ISSUE #7457: Day picker accessibility
references #7266
Improve accessibility of the day picker
- [x] the input can easily be operated by mouse
- [x] the input can be easily operated on touch devices, e.g.  smartphones, tablets or desktops/laptops with touch screens.
- [x]  the input can be easily operated by keyboard
- [x] users of all input methods should be aware of which dates are unavailable for selection

### PR DESCRIPTION
**Pull request by gestchild**
Fixes #7457

- replaces the react-day-picker with the mui one
- creates a new CalendarInput component for use with day picker
- adds aria-error and aria-errormessage and aria-describedby attributes to the input
- adds function and tests for checking selected date is an available date, when the library is open
- sets reducedAnimation property to true if user has prefers-reduced-motion set to true

### CODE REVIEW

File: catalogue/webapp/components/CalendarInput/CalendarInput.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import { forwardRef, useContext, useState } from 'react';
+// $FlowFixMe (tsx)
```
Comment (jamieparkinson) [inline catalogue/webapp/components/CalendarInput/CalendarInput.tsx:2]: We're in a TS file, aren't we?(!)

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx [hunk 2]
```diff
@@ -106,7 +117,7 @@ const RequestDialog: FC<RequestDialogProps> = ({
   currentHoldNumber,
 }) => {
   const { enablePickUpDate } = useToggles();
-  const [pickUpDate, setPickUpDate] = useState<Date | undefined>();
+  const [pickUpDate, setPickUpDate] = useState<Moment | null | undefined>(null);
```
Comment (jamieparkinson) [inline catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:120]: Why do we need `null` and `undefined`? If it's necessary can we have a comment about it?

File: catalogue/webapp/package.json [hunk 3]
```diff
@@ -14,9 +14,15 @@
   "dependencies": {
     "@babel/core": "^7.1.5",
     "@babel/preset-flow": "^7.12.1",
+    "@date-io/moment": "^2.11.0",
+    "@emotion/react": "^11.7.1",
+    "@emotion/styled": "^11.6.0",
```
Comment (jamieparkinson) [inline catalogue/webapp/package.json:19]: What are these for? Does mui need them?

File: catalogue/webapp/utils/dates.ts [hunk 4]
```diff
@@ -162,3 +162,35 @@ export function extendEndDate(params: {
     }
   }
 }
+
+export function isValidDate(params: {
```
Comment (jamieparkinson) [inline catalogue/webapp/utils/dates.ts:166]: Can we rename this to something that better describes what it does?  Seems like it's (a) not operating on just 1 date and (b) is Wellcome Collection-specific in that it considers closed days etc

**Review by jamieparkinson — COMMENTED** [review]
Can we have a screenshot of the new datepicker?

File: catalogue/webapp/components/RequestingDayPicker/RequestingDayPicker.tsx [hunk 5]
```diff
@@ -1,395 +1,136 @@
...
+  isValidDate,
 } from '@weco/catalogue/utils/dates';
 import { usePrismicData } from '@weco/common/server-data/Context';
 import {
   parseCollectionVenues,
   getVenueById,
 } from '@weco/common/services/prismic/opening-times';
+import DateAdapter from '@mui/lab/AdapterMoment';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import DatePicker from '@mui/lab/DatePicker';
```
Comment (jamieparkinson) [inline catalogue/webapp/components/RequestingDayPicker/RequestingDayPicker.tsx:23]: Think one of my comments got lost somehow - I'm a bit worried about the bundle size increase from these, have we looked at the loading in the date picker stuff [dynamically](https://nextjs.org/docs/advanced-features/dynamic-import) only when it's opened?

File: catalogue/webapp/utils/dates.ts (hunk 4 above)
Comment (gestchild) [reply catalogue/webapp/utils/dates.ts:166]: Yeah, it's checking whether the selected date falls within a range and is also not a day on which the library is closed.
Not sure how to capture that in a name? Does isRequestableDate work?

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx (hunk 2 above)
Comment (gestchild) [reply catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:120]: We did. Setting null stops the calendar from selecting a default value. But I don't think we want this behaviour, so I'm going to take that out.

File: catalogue/webapp/components/CalendarInput/CalendarInput.tsx (hunk 1 above)
Comment (gestchild) [reply catalogue/webapp/components/CalendarInput/CalendarInput.tsx:2]: we are! - copy and pasted, should have noticed

**Pull request closed without merging**